src/commands/revealResource.ts

```diff
@@ -18,12 +18,9 @@ export async function revealResource(context: IActionContext, arg: AppResource |
     context.telemetry.properties.resourceType = parseAzureResourceId(resourceId).provider.replace(/\//g, '|');
 
     try {
-        const subscriptionNode: SubscriptionTreeItem | undefined = await ext.appResourceTree.findTreeItem(`/subscriptions/${parseAzureResourceId(resourceId).subscriptionId}`, { ...context, loadAll: true });
-        const appResourceNode: AppResourceTreeItem | undefined = await subscriptionNode?.findAppResourceByResourceId(context, resourceId);
-        if (appResourceNode) {
-            // ensure the parent node loaded this AppResourceTreeItem
-            await appResourceNode.parent?.getCachedChildren(context);
-            await ext.appResourceTreeView.reveal(appResourceNode, { select: true, focus: true, expand: false });
+        const item: ResourceGroupsItem | undefined = await (ext.v2.api.resources.azureResourceTreeDataProvider as ResourceTreeDataProviderBase).findItemById(resourceId);
+        if (item) {
+            await ext.appResourceTreeView.reveal(item as unknown as AzExtTreeItem, { expand: false, focus: true, select: true });
```

Comment:
Same as in #481, why did this need to be cast as an `AzExtTreeItem`?

Reply:
`ext.appResourceTreeView` is still typed as `TreeView<AzExtTreeItem>`. We need to change the type and rename it. I'll do that in a follow up PR, as I'll also have to remove some old v1 stuff that relies on it.

Reply:
I thought we changed the interface to `TreeView<unknown>`

Reply:
Ah yeah. Maybe add a comment for that too.

Reply:
Will do. We need to cleanup the extension variables. There are a bunch of residual variables from v1 that probably can be removed, or at least need to be renamed.